openwisp_radius/tests/test_api.py

```diff
@@ -1441,14 +1441,21 @@ def test_cache(self):
         token_querystring = '?token={0}&uuid={1}'.format(rad.token, str(self.org.pk))
         post_url = '{}{}'.format(reverse('radius:authorize'), token_querystring)
         self.client.post(post_url, {'username': 'tester', 'password': 'tester'})
-        self.assertEqual(rad.token, cache.get(rad.organization.pk))
-        # test update
-        rad.token = '1234567'
-        rad.save()
-        self.assertEqual(rad.token, cache.get(rad.organization.pk))
-        # test delete
-        rad.delete()
-        self.assertEqual(None, cache.get(rad.organization.pk))
+        with self.subTest('Cache & token match'):
+            self.assertEqual(rad.token, cache.get(rad.organization.pk))
+        with self.subTest('Force changed corrupt cache returns 403'):
+            cache.set(rad.organization.pk, 'wrong-value')
+            response = self.client.post(
+                post_url, {'username': 'tester', 'password': 'tester'}
+            )
+            self.assertEqual(response.status_code, 403)
```

Comment:
Added a new testcase to make testing of the cache more robust.
If the value of cache is not as expected (here forcefullly changed), then it returns 403; this test would have reported the problem solved in this issue.